src/main/java/com/balako/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.balako.onlinebookstore.service.impl;
+
+import com.balako.onlinebookstore.dto.order.request.CreateOrderRequestDto;
+import com.balako.onlinebookstore.dto.order.request.UpdateOrderRequestDto;
+import com.balako.onlinebookstore.dto.order.response.OrderDto;
+import com.balako.onlinebookstore.dto.order.response.OrderItemDto;
+import com.balako.onlinebookstore.enums.Status;
+import com.balako.onlinebookstore.exception.EntityNotFoundException;
+import com.balako.onlinebookstore.mapper.OrderItemMapper;
+import com.balako.onlinebookstore.mapper.OrderMapper;
+import com.balako.onlinebookstore.model.Order;
+import com.balako.onlinebookstore.model.OrderItem;
+import com.balako.onlinebookstore.model.ShoppingCart;
+import com.balako.onlinebookstore.model.User;
+import com.balako.onlinebookstore.repository.cart.ShoppingCartRepository;
+import com.balako.onlinebookstore.repository.cartitem.CartItemRepository;
+import com.balako.onlinebookstore.repository.order.OrderRepository;
+import com.balako.onlinebookstore.repository.orderitem.OrderItemRepository;
+import com.balako.onlinebookstore.service.OrderService;
+import com.balako.onlinebookstore.service.UserService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final UserService userService;
+    private final ShoppingCartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderDto create(CreateOrderRequestDto requestDto) {
+        User user = userService.getCurrentAuthenticatedUser();
+        ShoppingCart cart = cartRepository.findByUserWithCartItems(user.getId()).orElseThrow(
+                () -> new EntityNotFoundException("Can't find cart by user email: "
+                        + user.getEmail()));
+
+        Order order = orderMapper.toModel(requestDto);
+        order.setUser(user);
+        order.setStatus(Status.getDefaultStatus());
```

Comment:
```suggestion
        order.setStatus(OrderStatus.getDefaultStatus());
```
Better rename the enum, because it's a bit confusing